### PR TITLE
workflows: fixes

### PR DESCRIPTION
**Pull request by jalavik**
A bundle of fixes that popped up during our INSPIRE labs testing.

Most notably is the ability to better extend the details page and "Need action" being a default selector of Holding Pen.

### CODE REVIEW

**Comment by jalavik** [comment]
@jirikuncar squashed and added some minor changes.

File: invenio/modules/workflows/models.py [hunk 1]
```diff
@@ -378,6 +385,9 @@ def get_formatted_data(self, of="hd"):
         """Get the formatted representation for this object."""
         from .registry import workflows
         try:
+            name = self.get_workflow_name()
+            if not name:
+                return ""
             workflow_definition = workflows[self.get_workflow_name()]
```
Comment (jirikuncar) [inline invenio/modules/workflows/models.py]: Can you reuse the `name` variable here?

File: invenio/modules/workflows/templates/workflows/details_base.html [hunk 2]
```diff
@@ -86,141 +86,165 @@
...
-                      {% if  object.id == bwobject.id %}
-                        <a href="#">{{ _('current') }} </a> -
-                        {% else %}
-                        <a href="{{ url_for('holdingpen.details', objectid=object.id) }}" >{{ object.id }}</a> -
-                      {% endif %}
-                      <span class="label label-{{ object._class }}">{{ object.message }}</span>
-                     {{ pretty_date(object.modified) }}
-                    <hr>
-                    {% endfor %}
+                      {% for key, value in info.items() %}
```
Comment (jirikuncar) [inline invenio/modules/workflows/templates/workflows/details_base.html]: why aren't you using `table` rows here if https://github.com/inveniosoftware/invenio/pull/2586/files#diff-05794fe86f3fba5b0c0d667b0b7cf7a7R201 defines classes used for table? IMHO compact table would be nicer here.

File: invenio/modules/workflows/templates/workflows/details_base.html [hunk 3]
```diff
@@ -86,141 +86,165 @@
...
+                <div class="panel-heading">
+                  <h3 class="panel-title">{{ _('History') }}</h3>
+                </div>
+                <div class="panel-body">
+                  <h6 class="pull-right">
+                    <a role="button" style="cursor: pointer" data-toggle="collapse" data-target="#History">{{ _('More') }}<span class="caret"></span></a>
+                  </h6>
+                  <div class="row"></div>
+                  <div id="History" class="collapse">
+                    <div class="table table-striped">
```
Comment (jirikuncar) [inline invenio/modules/workflows/templates/workflows/details_base.html]: similarly here ... `class="table ..."` and no table around

File: invenio/modules/workflows/models.py (hunk 1 above)
Comment (jalavik) [inline invenio/modules/workflows/models.py]: Indeed! Thanks.

File: invenio/modules/workflows/templates/workflows/details_base.html (hunk 2 above)
Comment (jalavik) [inline invenio/modules/workflows/templates/workflows/details_base.html]: I agree this is strange. Must be something leftover. I will patch it. Thanks!

File: invenio/modules/workflows/templates/workflows/details_base.html (hunk 3 above)
Comment (jalavik) [inline invenio/modules/workflows/templates/workflows/details_base.html]: Indeed. Will fix

File: invenio/modules/workflows/engine.py [hunk 4]
```diff
@@ -461,35 +460,13 @@ def processing_factory(objects, self):
...
-                    # stopped
-                    if type(e) == WorkflowHalt:
-                        reraise(*sys.exc_info())
-                    else:
-                        raise WorkflowHalt(e)
-                except (WorkflowErrorClient, WorkflowError, Exception) as e:
-                    # unless instructed otherwise.
+                    raise
+                except Exception as e:
                     msg = "Error: %r\n%s" % (e, traceback.format_exc())
```
Comment (dset0x) [inline invenio/modules/workflows/engine.py]: `msg` is not use anywhere anymore.

**Comment by jalavik** [comment]
Fixes kwalitee issues and failing test. Just need reviewer on  017ff1c and 185ab5b.

**Comment by jalavik** [comment]
All is well. :+1:

**Pull request merged**